addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-mount-service.js

```diff
@@ -667,16 +667,53 @@ class EnvironmentMountService extends Service {
     return policyDoc.Statement;
   }
 
+  async _getPolicyDoc(possiblePolicyNames, roleName, workspaceRoleArn, iamClient) {
+    const iamService = await this.service('iamService');
+    const policyNamePrefix = `analysis-${workspaceRoleArn.split('-')[1]}`;
+
+    const policyDocsInfos = await Promise.all(
+      _.map(possiblePolicyNames, async possiblePolicyName => {
+        let policyDocStr = '';
+        let studyDataPolicyName = '';
+        try {
+          studyDataPolicyName = `${policyNamePrefix}-${possiblePolicyName}`;
+          policyDocStr = await iamService.getRolePolicy(roleName, studyDataPolicyName, iamClient).PolicyDocument;
+        } catch (err) {
```

Comment:
The "getRolePolicy" method internally already handles `NoSuchEntity` and `ResourceNotFoundException` exceptions and returns empty object so you will not get any error here when the policy does not exist.

Reply:
Got you. Thanks for mentioning that. Made the changes per our discussion.